source/common/http/conn_manager_impl.cc

```diff
@@ -367,23 +367,16 @@ void ConnectionManagerImpl::resetAllStreams(
     // codec but there are no easy answers and this seems simpler.
     auto& stream = *streams_.front();
     stream.response_encoder_->getStream().removeCallbacks(stream);
-    stream.onResetStream(StreamResetReason::ConnectionTermination, absl::string_view());
+    if (!stream.response_encoder_->getStream().responseDetails().empty()) {
+      stream.filter_manager_.streamInfo().setResponseCodeDetails(
+          stream.response_encoder_->getStream().responseDetails());
+    } else if (!details.empty()) {
+      stream.filter_manager_.streamInfo().setResponseCodeDetails(details);
+    }
```

Comment:
qq -- this fixes these three cases, but there's no "catch all" for if the response details are still empty. This in the hopes that the ASSERT will reveal anything missing, is that right?

Reply:
Yep, the assert should take care of it.
The only place we set empty details is the idle timeout, which should only fire if there are no streams to terminate.   If others add other empty strings it's annoying and will trigger a debug assert for "you are doing the wrong thing" but given it's just missing debug info, not mission critical, I think that's fine.